testgridshot

```diff
@@ -0,0 +1,268 @@
+#+   stub on standard out. This stub is intended to be copy & pasted into a issue
+#+   comment for the release cutting GitHub issue.
+#+
+#+ Configuration:
+#+   Configuration can be passed in via the environmnet. The following
+#+   variables are available:
+#+
+#+   BOARDS: [default: "blocking informing"]
+#+     Change which boards you are interested in
+#+
+#+   STATES: [default: "FAILING"]
+#+     Change the tests you want to screenshot
+#+     Available states: FAILING, FLAKY, PASSING
+#+
+#+   BLOCK_WIDTH: [default: 30]
+#+     The width of the individual testgrid red & green block.
+#+     Influences on how many test runs will be shown on the screenshot.
+#+
+#+   WIDTH/HEIGHT: [default: WIDTH=3000/HEIGHT=2500]
+#+     The width and height of the generated screenshot in pixel.
+#+
+#+   RETRY_COUNT: [default: 3]
+#+     Change how often we should retry when talking to (curl'ing) APIs.
+#+
+#+   RETRY_SLEEP: [default: 2]
+#+     Change the amount of time we want to wait between consecutive API call
+#+     retries.
+#+
+#+   Example:
+#+     $ BOARDS='all informing' STATES='FAILING FLAKY' BLOCK_WIDTH=10 ./testgridshot 1.15
+#+         Creates dense screenshots of all the 'sig-release-1.15-{all,informing}'
+#+         boards which are either failing or flaking
+#+
+#+ How it works:
+#+   General flow:
+#+   - Get the dasbhoard URLs we are interested in from testgrid's 'summary'
+#+     endpoint
+#+   - Use 'render-tron.appspot.com' to create screenshots of the testgrid
+#+     boards
+#+   - Upload all the screenshots to 'vgy.me'
+#+   - Print a markdown, which links to the images on 'vgy.me', stub on StdOut
+#+   Remarks:
+#+   - Both 'vgy.me' and 'render-tron.appspot.com' might go away anytime
+#+   - The images are not served by 'vgy.me' directly if used in a GitHub issue,
+#+     but by GitHub's camo service
+#+
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+readonly STATE_FAILING='FAILING'
+readonly STATE_FALKING='FLAKY'
+readonly STATE_PASSING='PASSING'
+
+readonly BOARDS="${BOARDS:-blocking informing}"
+readonly STATES="${STATES:-${STATE_FAILING}}"
+readonly BLOCK_WIDTH="${BLOCK_WIDTH:-30}"
+readonly WIDTH="${WIDTH:-3000}"
+readonly HEIGHT="${HEIGHT:-2500}"
+readonly RETRY_COUNT="${RETRY_COUNT:-3}"
+readonly RETRY_SLEEP="${RETRY_SLEEP:-2}"
+
+readonly TESTGRID='https://testgrid.k8s.io'
+readonly RENDER_TRON='https://render-tron.appspot.com/screenshot'
+readonly UPLOAD_URL='https://vgy.me/upload'
+readonly ISSUE_STUB_SUFFIX='issue.'
+
+get_tests_by_status() {
+  local status="$1"
+  shift
+
+  local board
+
+  for board in "$@"
+  do
+    run_curl --retry 0 "${TESTGRID}/${board}/summary" 2>/dev/null \
+      | jq --arg status "$status" -r '
+        to_entries[]
+          | select(.value.overall_status==$status)
+          | .value.dashboard_name + "#" + .key
+      '
+  done
+}
+
+urlencode() {
+  echo "$1" | jq -sRr @uri
+}
+
+gen_file_name() {
+  echo "${1//[^a-zA-Z0-9]/_}"
+}
+
+run_curl() {
```

Comment:
```suggestion
curl_with_retry() {
```

Reply:
Let's be descriptive in the function name about why we're using this (because it includes retries) instead of regular `curl`, and change the references through this script.